File: risk/models/cards.py
"""Card and turn phase models for Risk game."""

from enum import auto, Enum

from pydantic import BaseModel


class CardType(Enum):
    """Types of Risk territory cards."""

    INFANTRY = auto()
    CAVALRY = auto()
    ARTILLERY = auto()
    WILD = auto()


class TurnPhase(Enum):
    """Phases within a single player's turn."""

    REINFORCE = auto()
    ATTACK = auto()
    FORTIFY = auto()


class Card(BaseModel):
    """A Risk territory card (or wild card)."""

    territory: str | None
    card_type: CardType

File: mobile/lib/engine/models/cards.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'cards.freezed.dart';
part 'cards.g.dart';

enum CardType { infantry, cavalry, artillery, wild }

enum TurnPhase { reinforce, attack, fortify }

@freezed
abstract class Card with _$Card {
  const factory Card({
    String? territory,
    required CardType cardType,
  }) = _Card;

  factory Card.fromJson(Map<String, dynamic> json) => _$CardFromJson(json);
}
